include/SDL3/SDL_filesystem.h

```diff
@@ -236,6 +236,125 @@ typedef enum
+     *
+     * \param fs The SDL_FSops object to use.
+     * \param path the path in the SDL_FSops to open
+     * \param mode a fopen-style "mode" string of how to access the path.  !!! FIXME: maybe just offer read/write/append enums?
+     * \return a RWops for the path specified.
+     */
+    SDL_RWops *(SDLCALL * open)(SDL_FSops *fs, const char *path, const char *mode);
+
+    /* list all files in a directory. Each file is passed through a callback until we're done or the callback returns <= 0. */
+    int (SDLCALL * enumerate)(SDL_FSops *fs, const char *path, SDL_EnumerateCallback cb, void *userdata);
+
+    /* delete/unlink/rmdir a path. Will not remove non-empty directories! */
+    int (SDLCALL * remove)(SDL_FSops *fs, const char *path);
+
+    /* Create a directory. */
+    int (SDLCALL * mkdir)(SDL_FSops *fs, const char *path);
+
+    /* Obtain basic path metadata. */
+    int (SDLCALL * stat)(SDL_FSops *fs, const char *path, SDL_Stat *stat);
+
+    /* Close the filesystem and free its resources. */
+    void (SDLCALL * closefs)(SDL_FSops *fs);
+
+    /* Used by the SDL_FSops for private data. Don't touch! */
+    void *opaque;
+
+    /* Everything has properties! :) */
+    SDL_PropertiesID props;
+};
+
+
+/*
+ * Create an SDL_FSops that represents the native filesystem, treating `basedir` as
+ * the root of the tree. A NULL basedir is valid and gives you the actual root of the
+ * native filesystem.
+ *
+ * Enumerating root on Windows will give you available drive letters like they were directories,
+ * almost everything else assumes the root is a Unix "/" thing.
+ *
+ * Destroy this when done with SDL_DestroyFilesystem().
+ */
+extern DECLSPEC SDL_FSops * SDLCALL SDL_CreateFilesystem(const char *basedir);
+
+
+/* Use the filesystem... */
+extern DECLSPEC SDL_RWops * SDLCALL SDL_FSopen(SDL_FSops *fs, const char *path, const char *mode);
+extern DECLSPEC int SDLCALL SDL_FSenumerate(SDL_FSops *fs, const char *path, SDL_EnumerateCallback cb, void *userdata);
+extern DECLSPEC int SDLCALL SDL_FSremove(SDL_FSops *fs, const char *path);
+extern DECLSPEC int SDLCALL SDL_FSmkdir(SDL_FSops *fs, const char *path);
+extern DECLSPEC int SDLCALL SDL_FSstat(SDL_FSops *fs, const char *path, SDL_Stat *stat);
+
+/* returns `props` field of `context`, creating an SDL_PropertiesID for that field, if it doesn't exist yet. */
+extern DECLSPEC SDL_PropertiesID SDLCALL SDL_GetFSProperties(SDL_FSops *context);
+
+/**
+ * if fs is not NULL, this will call fs->closefs() (if closefs is not NULL), destroy fs->props, and then SDL_free(fs). You can use this
+ * with your own SDL_FSops implementations if that sequence of operations is correct.
+ */
+extern DECLSPEC void SDLCALL SDL_DestroyFilesystem(SDL_FSops *fs);
+
+
+/* some helper functions ... */
+
+/* returns "\\" or "/" etc. Read-only static data, don't free or modify! */
+extern DECLSPEC const char * SDLCALL SDL_GetFilePathSeparator(void);
```

Comment:
I don't think we need this, we can assume '/' works for all modern operating systems.

Reply:
IIRC windows still requires `\` in some cases, and it's still considered the canonical separator. You can [prepend `\\?\` to a path to skip normalization](https://learn.microsoft.com/en-us/dotnet/standard/io/file-path-formats#skip-normalization). The docs aren't exactly clear on whether that includes the `/` to `\\` conversion, that's something to check.

Reply:
Removed in 5a96b8c3e9c424c79e9d890efc9301b1ccdfe900